### PR TITLE
show all detected fields in prebuilt predict result

### PR DESCRIPTION
**Pull request by yongbing-chen**
show all detected fields in prebuilt predict result

### CODE REVIEW

**Review by starain-pactera — APPROVED** [review]

File: src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx [hunk 1]
```diff
@@ -671,37 +671,30 @@ export class PrebuiltPredictPage extends React.Component<IPrebuiltPredictPagePro
...
+                            result[`${itemName}: ${key}`] = item.valueObject[key];
+                        });
+                    }
+                    else {
+                        result[itemName] = item;
+                    }
+                }
+
                 if (field.valueArray) {
                     if (field.valueArray.length === 1) {
```
Comment (buddhawang) [inline src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx:686]: suggestion: I would prefer simplify a bit more as the if-else are only different in the first itemName. We can easily achieve the same result with field.valueArray.forEach and checking the item index in the foreach.

File: src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx [hunk 2]
```diff
@@ -671,37 +671,30 @@ export class PrebuiltPredictPage extends React.Component<IPrebuiltPredictPagePro
 
             const extendPredictionItem = (key, field) => {
                 const result = {};
+                const updateFieldValueToResult = (item, itemName) => {
+                    if (item.valueObject) {
```
Comment (buddhawang) [inline src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx:675]: suggestion: ideally we should have different function for handling different structured item.valueXxx. But that might be out of this PR's scope IMHO.

File: src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx (hunk 1 above)
Comment (yongbing-chen) [reply src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx:686]: like this?
```
                    field.valueArray.forEach((item, index) => {
                        const itemName = field.valueArray.length === 1 ? key : `${key} ${index + 1}`;
                        updateFieldValueToResult(item, itemName);
                    });
```

**Review by buddhawang — APPROVED** [review]
Left 2 minor comments, others looks good to me, thanks!

File: src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx (hunk 1 above)
Comment (buddhawang) [reply src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx:686]: yes :)

**Review by buddhawang — APPROVED** [review]

**Review by starain-pactera — APPROVED** [review]

**Pull request merged**